ophyd/signal.py

```diff
@@ -95,7 +102,7 @@ def __init__(
         self,
         *,
         name,
-        value=0.0,
+        value=DEFAULT_EPICSSIGNAL_VALUE,
```

Comment:
If we move this here I suspect we will have to hoist the logic to handle this from EpicsSignal up to the base class.

Reply:
well, it looks like `DEFAULT_EPICSSIGNAL_VALUE` is more used as a sentinel value to indicate if a default has been provided or not - it is currently already used in `Signal` class, see line 475:

``` 
if self._readback is DEFAULT_EPICSSIGNAL_VALUE:
            val = self.get()
```

or derived classes which do not have an explicit relation with EPICS (i.e not deriving from `EpicsSignalBase`).

I propose to rename it, then ? Like `UNSET_VALUE` or `UNSPECIFIED_VALUE` ?

It seems `None` is considered as a potential ok value. Is it the case ? Otherwise we could initialize with `None`.

In `describe()`, I think the key point is to really read the value if it is not specified. Any way to do it like this is fine for me.

What do you think ?

Reply:
Oh, if we are already using it is the base class than I'm 50/50 on adding an alias (the hassle of anyone downstream using the old name needing to update is not worth it to remove the old name) vs just using it.

The concern I still have is that in `Signal.get` we just return `self._readback` which means that there is a chance of returning `DEFAULT_EPICSSIGNAL_VALUE` to the user (which we definitely should not do).  The minimal fix is probably to raise in `Signal.get` if the value has never been set.

Reply:
Wouldn't it be better to initialize to `None` and to return `None` if it has not been set? I am very new to Ophyd, so I do not know the logic/philosophy behind reading signals and why `DEFAULT_EPICSSIGNAL_VALUE` was there at first place

Reply:
Otherwise I can make an alias and raise in `get()` as you suggest, no problem 👍🏻 

Tell me what you prefer

Reply:
tl;dr:  I would prefer the alias + raise in `get` path as the minimal change.  

pyepics (the default wrapping we use to talk channel access over in Python) returns `None` when it fails so defaulting to `None` means we could not tell the difference between "failed to read" vs "never read".  The logic was to have a unique sentinel that would never come to use from pyepics and would never be passed to us by the user so we could tell if we had ever been told a value for this signal. Basically detect the start up transient and allows lazy connection (each PV is its own connection dance with a server and if we never read it then it is better to never try to connect).

Going with a custom default object also helps makes clear that it is "default but invalid" where as `None` is frequently "default and valid" or "placeholder to tell the system to do the default thing" and I think that is an important distinction here.

I'm a bit fuzzy remember choices we made probably 8 years ago now, but I am biased to trusted my previous self 🤣